frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { Form, ActionGroup, ButtonVariant, Button } from '@patternfly/react-core';
+import { ButtonBar, useAccessReview } from '@console/internal/components/utils';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import AppSection from '@console/dev-console/src/components/import/app/AppSection';
+import { FirehoseList } from '@console/dev-console/src/components/import/import-types';
+import CronJobSection from './event-sources/CronJobSection';
+import SinkSection from './event-sources/SinkSection';
+import { EventSources } from './import-types';
+import {
+  EventSourceCronJobModel,
+  EventSourceSinkBindingModel,
+  EventSourceApiServerModel,
+  EventSourceCamelModel,
+  EventSourceKafkaModel,
+} from '../../models';
+import EventSourcesSelector from './event-sources/EventSourcesSelector';
+import * as apiServerSourceImg from '../../imgs/logos/apiserversource.png';
+import * as camelSourceImg from '../../imgs/logos/camelsource.svg';
+import * as containerSourceImg from '../../imgs/logos/containersource.png';
+import * as cronJobSourceImg from '../../imgs/logos/cronjobsource.png';
+import * as kafkaSourceImg from '../../imgs/logos/kafkasource.svg';
+
+interface OwnProps {
+  services: FirehoseList;
+  projects: FirehoseList;
+}
+
+const useKnativeEventingAccess = (model): boolean => {
+  const canCreateEventSource = useAccessReview({
+    group: model.apiGroup,
+    resource: model.plural,
+    namespace: getActiveNamespace(),
+    verb: 'create',
+  });
+  return canCreateEventSource;
+};
+
+const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
+  values,
+  errors,
+  handleSubmit,
+  handleReset,
+  status,
+  isSubmitting,
+  dirty,
+  projects,
+  services,
+}) => {
+  const eventSourceList = [
```

Comment:
This whole logic should move out of the component render scope. Preferably utils file.

Reply:
Could just be a function `getEventSources` to be able to handle future use cases where the list is dynamic.